ConsoleBank/ConsoleBank/Accounting/Account/Transfer/AccountTransferService/AccountTransferService.cs

```diff
@@ -62,50 +55,44 @@ public async Task Transfer(AccountTransferOptions options)
             });
         }
 
-        private async Task RequestTransfer(AccountTransferOptions options,
-            Guid transactionId, decimal withdrawAmount, decimal acquireAmount)
+        private async Task TryTransfer(AccountTransferOptions options,
+            decimal withdrawAmount, decimal acquireAmount, Guid transactionId)
         {
-            try
-            {
-                await _accountAquiring.Withdraw(options.FromAccount, withdrawAmount, transactionId);
-                await _accountAquiring.Acquire(options.ToAccount, acquireAmount, transactionId);
-            }
-            catch (Exception e)
+            var transactionInfo = await _transactionTransferService.GetTransactionInfoById(transactionId);
+
+            if (transactionInfo.TransferStatus == TransactionTransferStatus.InProgress
+                && transactionInfo.TransactionId == transactionId)
             {
-                await CancelTransfer(transactionId);
-                throw new Exception($"Transaction was canceled because {e.Message}");
+                try
+                {
+                    await TryWithdraw(options, withdrawAmount);
+                    await TryAcquire(options, acquireAmount);
+
+                }
+                catch (Exception e)
+                {
+                    await _transactionTransferService.Cancel(transactionInfo.TransactionId, e.Message);
+                    throw new Exception($"Can't make a transfer because: {e.Message}");
```

Comment:
лучше писать так, тогда мы сохраним оригинальный stacktrace
```suggestion
                    throw;
```